lib/data/external/GCP/GcpApis/copyObject.js

```diff
@@ -0,0 +1,112 @@
+const async = require('async');
+const request = require('request');
+const { errors } = require('arsenal');
+
+const { logger, JsonError, checkSourceString } = require('../GcpUtils');
+const { logHelper } = require('../../utils');
+
+/**
+ * copyObject - minimum required functionality to perform object copy
+ * for GCP Backend
+ * @param {object} params - update metadata params
+ * @param {string} params.Bucket - bucket name
+ * @param {string} params.Key - object key
+ * @param {string} param.CopySource - source object
+ * @param {function} callback - callback function to call with the copy object
+ * result
+ * @return {undefined}
+ */
+function _copyObject(params, callback) {
+    const { CopySource } = params;
+    if (!checkSourceString(CopySource)) {
+        if (CopySource) {
+            return callback(errors.InvalidArgument);
+        }
+        return callback(errors.MissingParameter);
+    }
+    const sourceArray = CopySource.split('/');
+    const sourceBucket = sourceArray[0] ? sourceArray[0] : sourceArray[1];
```

Comment:
Can `sourceArray[0]` ever be falsy'?

Reply:
Yes. Ex. `/a/b`, the split will generate array [ '', 'a', 'b' ]

Reply:
Ok, did not realize leading slash is handled.